### ISSUE TITLE
[BUG] `go install` instructions do not work 

### ISSUE THREAD

**Issue by whi-tw** [StackExchange/dnscontrol#3056]
# The Bug

The installation instructions state that `dnscontrol` can be installed with `go install`: https://github.com/StackExchange/dnscontrol/blob/086905241969ebd7cf28c1a43157b81291da4eb2/documentation/getting-started.md?plain=1#L42-L44

However, because of the `replace` directives, this doesn't actually work:

```shell
❯ go version
go version go1.22.5 linux/amd64
❯ go install github.com/StackExchange/dnscontrol/v4@latest
go: github.com/StackExchange/dnscontrol/v4@latest (in github.com/StackExchange/dnscontrol/v4@v4.12.5):
        The go.mod file for the module providing named packages contains one or
        more replace directives. It must not contain directives that would cause
        it to be interpreted differently than if it were the main module.
```

This is not a recently introduced issue: https://github.com/golang/go/issues/44840, so either I'm missing something, or people haven't attempted this in a while!

**Comment by cafferata** [comment]
The replace directive was introduced in GitHub pull request https://github.com/StackExchange/dnscontrol/pull/3002 [here](https://github.com/StackExchange/dnscontrol/pull/3002/commits/7cfe882d938e9f348106c7b5e221a91a16e66a3a#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R7).

> replace github.com/cloudflare/cloudflare-go => ./pkg/cloudflare-go 

cc: @tlimoncelli

**Comment by whi-tw** [comment]
Agreed - updating the docs seems like the best approach for now. I suppose the fact I'm the first person to raise it after two months suggests that this isn't inconveniencing (m)any people too much!

**Comment by jauderho** [comment]
+1 from me.

I noticed this just the other day when I was trying to use `go install` too.

It appears to be a long standing issue with Go:
* https://github.com/golang/go/issues/44840
* https://www.reddit.com/r/golang/comments/187uok8/the_issue_about_replace_directives_in_the_main/

Looking forward to when the cloudflare-go changes get merged upstream.

**Comment by tlimoncelli** [comment]
I'm optimistic Cloudflare will accept my PR: https://github.com/cloudflare/cloudflare-go/pull/2833

**Comment by cafferata** [comment]
You were absolutely right! GitHub pull request https://github.com/cloudflare/cloudflare-go/pull/2833 has been merged! 🚀 👏🏻

**Comment by jauderho** [comment]
I just tried building dnscontrol using the `cloudflare-go` and removing the replace statement in go.mod. 

Getting the following error message:
```
github.com/StackExchange/dnscontrol/v4/providers/cloudflare
# github.com/StackExchange/dnscontrol/v4/providers/cloudflare
providers/cloudflare/rest.go:391:115: too many arguments in call to c.cfClient.DeleteRulesetRule
	have (context.Context, *"github.com/cloudflare/cloudflare-go".ResourceContainer, string, string)
	want (context.Context, *"github.com/cloudflare/cloudflare-go".ResourceContainer, "github.com/cloudflare/cloudflare-go".DeleteRulesetRuleParams)
```

**Comment by cafferata** [comment]
`DeleteRulesetRule()` has been released in [v0.102.0](https://github.com/cloudflare/cloudflare-go/releases/tag/v0.102.0). 🥳

**Comment by jauderho** [comment]
Confirmed that `go install github.com/StackExchange/dnscontrol/v4@e8eca6a` compiles just fine.
